default/python/AI/EspionageAI.py

```diff
@@ -45,7 +53,10 @@ def colony_detectable_by_empire(planet_id, species_name=None, empire_id=ALL_EMPI
     # The future_stealth_bonus can be used if the AI knows it has researched techs that would grant a stealth bonus to
     # the planet once it was colonized/captured
 
-    empire_detection = get_empire_detection(empire_id)
+    if isinstance(empire, int):
+        empire_detection = get_empire_detection(empire)
```

Comment:
you can use `get_max_empire_detection([impire])` and make `get_empire_detection` private

Reply:
Please move this code close to its usage.

Reply:
> Please move this code close to its usage.

OK, moved

> you can use get_max_empire_detection([impire]) and make get_empire_detection private

While that could be possible with some reworking of things, as has now been clarified in the docstrings, ```get_max_empire_detection()``` excludes the current AI empire-- you could think of it as ```get_max_non_self_empire_detection()``` but that seems to me an unnecessarily cumbersome name.   So things seem simpler/better to me as-is.